lib/ssz_ex.ex

```diff
@@ -495,15 +593,67 @@ defmodule LambdaEthereumConsensus.SszEx do
   defp remove_trailing_bit(<<0::7, 1::1>>), do: <<0::0>>
   defp remove_trailing_bit(<<0::8>>), do: <<0::0>>
 
-  defp pack(value, size) when is_integer(value) and value >= 0 do
-    pad = @bits_per_chunk - size
-    <<value::size(size)-little, 0::size(pad)>>
+  defp size_of(:bool), do: @bytes_per_boolean
+
+  defp size_of({:int, size}), do: size |> div(@bits_per_byte)
+
+  defp chunk_count({:list, {:int, size}, max_size}) do
+    size = size_of({:int, size})
+    (max_size * size + 31) |> div(32)
   end
 
-  defp pack(value) when is_boolean(value) do
-    case value do
-      true -> <<1::@bits_per_chunk-little>>
-      false -> <<0::@bits_per_chunk>>
+  defp chunk_count({:list, :bool, max_size}) do
+    size = size_of(:bool)
+    (max_size * size + 31) |> div(32)
+  end
+
+  defp pack_basic_type_list(list, schema) do
+    list
+    |> Enum.reduce(<<>>, fn x, acc ->
```

Comment:
We should replace this with a list + `Enum.join` (in a following PR)

Reply:
Does `Enum.join()` take functions?

Reply:
No. I mean replacing the binary with a list of binaries. That way you can avoid expensive string concatenations while iterating, and then at the end just use `Enum.join`.

But this case shouldn't be too bad in terms of performance, since we append the data in the accumulator (the VM seems to have optimizations for this kind of case).

Reply:
Gotcha